Core/src/org/sleuthkit/autopsy/datamodel/CreateSleuthkitNodeVisitor.java

```diff
@@ -98,7 +104,13 @@
 
     @Override
     public AbstractContentNode<? extends Content> visit(BlackboardArtifact art) {
-        return new BlackboardArtifactNode(art);
+        try {
+            return new BlackboardArtifactNode(BlackboardArtifactNodeKey.createNodeKey(art));
+        } catch (TskCoreException ex) {
+            logger.log(Level.SEVERE, String.format("Failed to creatNodeKey for artifact (%d)", art.getId()), ex);
```

Comment:
@rcordovano 
I mentioned in my email on Friday that CreateSleuthkitNodeVisitor will cause nodes to be created on the EDT.  I believe it will be used to create some of the newer nodes on the EDT, host, person, It could be potentially used to create BlackboardARtifactNodes on the EDT, thought I haven't found a place where this is happening.

Reply:
You were absolutely correct in your email last Friday. The "data-full key" strategy and the use of CreateSleuthkitNodeVisitor to create Content objects for use as node keys are incompatible. We could bypass CreateSleuthkitNodeVisitor as a key creation mechanism, but the way the data sources sub tree is structured would have to be rewritten and frankly we would no longer have any use for CreateSleuthkitNodeVisitor.